# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1381687055 -7200
#      Sun Oct 13 19:57:35 2013 +0200
# Node ID 7ec70c575ad64c410f18544605edf64745afe3b8
# Parent  993ea21b7e03dcbb26f407d3789cb822c30ed5e1
* octave-gui.cc: set the encoding of qt strings to utf8

diff --git a/libgui/src/octave-gui.cc b/libgui/src/octave-gui.cc
--- a/libgui/src/octave-gui.cc
+++ b/libgui/src/octave-gui.cc
@@ -20,16 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <QtGui/QApplication>
+#include <QTextCodec>
 #include <QTranslator>
 
 #include <iostream>
 
 #include <unistd.h>
 #include <fcntl.h>
 
 #if defined (HAVE_SYS_IOCTL_H)
@@ -92,16 +93,19 @@ dissociate_terminal (void)
 int
 octave_start_gui (int argc, char *argv[], bool fork)
 {
   if (fork)
     dissociate_terminal ();
 
   QApplication application (argc, argv);
 
+  // Set the codec for all strings
+  QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
+
   // install translators for the gui and qt text
   QTranslator gui_tr, qt_tr, qsci_tr;
   resource_manager::config_translators (&qt_tr,&qsci_tr,&gui_tr);
   application.installTranslator (&qt_tr);
   application.installTranslator (&qsci_tr);
   application.installTranslator (&gui_tr);
 
   while (true)
